### ISSUE TITLE
`width_heuristics` item not being parsed

### ISSUE THREAD

**Issue by chrisvittal** [rust-lang/rustfmt#4170]
I have a rustfmt.toml like so:
```toml
width_heuristics = "Max"
```
And when I run rustfmt, I get the following error
```
Error: Decoding config file failed:
invalid type: string "Max", expected struct WidthHeuristics for key `width_heuristics`
Please check your config file.
```

Version `rustfmt 1.4.14-nightly (a5cb5d2 2020-04-14)`

**Comment by calebcartwright** [comment]
Apologies for the confusion @chrisvittal.

Since you are using the released rustfmt 1.4.14 version, you'll need to use the `use_small_heuristics` config option.

As part of the upcoming rustfmt 2.0 release we're refactoring the width heuristics configuration options to give users more granular control, and the latest Configuration documentation (https://github.com/rust-lang/rustfmt/blob/master/Configurations.md and http://rust-lang.github.io/rustfmt/) is reflective of those rustfmt 2.0 changes.

rustfmt 2.0 (and new/updated config opts like `width_heuristics`) are currently only available when building rustfmt from source.

**Comment by marcospb19** [comment]
@calebcartwright I'm confused about https://github.com/rust-lang/rustfmt/blob/master/Configurations.md#width_heuristics

Are the docs wrong? Should it be `use_small_heuristics`?

**Comment by calebcartwright** [comment]
> Are the docs wrong? Should it be use_small_heuristics

No, and no

As alluded to above, and mentioned in various places within the documentation (including in that very same Configurations file)

https://github.com/rust-lang/rustfmt/blob/master/Configurations.md#configuration-options

> Note that the below list reflects the configuration options available on the latest version of rustfmt in source control. Some newer options may not be available yet in a released version of rustfmt.

> For version- and channel-specific configurations, please visit https://rust-lang.github.io/rustfmt/.

Unless you are compiling rustfmt from the latest commit in source control, you don't want to look at the latest commit of the Configurations.md file in source control.

There are always multiple versions of rustfmt being leveraged by different users, and the available configuration options are tightly coupled to the version of rustfmt. I.e. a new config option `foo` added in rustfmt 1.4.34 is only available/usable from v1.4.34 onward, and it's not available in versions < v1.4.34. Accordingly, you always want to look at the version of the configuration option documentation that aligns with the version of rustfmt you are actually using; you don't want to look at the docs for the latest and greatest in source control when you're not using/building rustfmt from source control.

Would strongly encourage everyone to leverage the documentation site for configuration options: https://rust-lang.github.io/rustfmt/ where it's much easier to map up the doc version with your rustfmt version, as opposed to looking at source control.
